crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -48,6 +57,13 @@ impl PartitionProducer {
         last_error: Arc<RwLock<Option<ProducerError>>>,
         #[cfg(feature = "stats")] client_stats: Arc<ClientStats>,
     ) -> Self {
+        #[cfg(feature = "otel-metrics")]
+        let meter = global::meter("producer");
+        #[cfg(feature = "otel-metrics")]
+        let records_counter = meter.u64_counter("fluvio.client.records").init();
```

Comment:
this should be more like: `fluvio.producer.records`.

Reply:
I was looking at convention and examples. https://opentelemetry.netlify.app/docs/reference/specification/metrics/semantic_conventions/database-metrics/

Since the produce and consume will have the same counters. The convention seems to be to use an attribute to distinguish produce and consume. See line 275

Also see `system.network.packets` at https://opentelemetry.netlify.app/docs/reference/specification/metrics/semantic_conventions/system-metrics/. `transmit` and `receive` is the value of the `direction` attribute.

Reply:
See http metrics which seems to jive more with our semantics:
https://opentelemetry.netlify.app/docs/reference/specification/metrics/semantic_conventions/http-metrics/
As well as Kafka:
https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/instrumentation/kafka/

Reply:
so following that it should be something like `messaging.fluvio.producer.outgoing`. etc

Reply:
updated